services/ymax-planner/test/engine.test.ts

```diff
@@ -52,149 +345,234 @@ const mockAsset = (
   });
 };
 
-test('ignore additional balances', t => {
-  const usdc = mockAsset('USDC');
-  const { denom, brand } = usdc;
+const depositAsset = mockAsset('USDC');
+const { boardId: depositBoardId, brand: depositBrand } = depositAsset;
+const { boardId: feeBoardId, brand: feeBrand } = mockAsset('Fee');
+const defaultMarshallerEntries: Pick<Map<string, unknown>, 'get'> = new Map([
+  [depositBoardId, depositBrand],
+  [feeBoardId, feeBrand],
+]);
+const defaultMarshaller = boardSlottingMarshaller<string>(
+  slot => defaultMarshallerEntries.get(slot) || Fail`Unknown slot ${slot}`,
+);
 
-  const balances = [
-    { amount: '50', denom },
-    { amount: '123', denom: 'ubld' },
-  ];
+let caipAddressCount = 0;
+const fakePortfolioKit = async ({
```

Comment:
ah. good. so we're now at the point where the planner package simulates/mocks the contract and the contract package mocks the planner. I have been looking forward to that since Palo Alto.